subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/metadata/ModulePath.kt

```diff
@@ -0,0 +1,11 @@
+package com.avito.instrumentation.impact.metadata
+
+data class ModulePath(val path: String) {
+    init {
+        require(path.matches(validModulePath)) { "$path is not a valid gradle module path" }
+    }
+}
+
+fun ModulePath.toFileName() = path.drop(1).replace(":", "+")
```

Comment:
Can paths clash after dropping a caught group?

Reply:
Could we use Gradle API? We may get project dir by a path. Parsing string, dropping ':' looks fragile.
Not a blocker

Reply:
i used https://github.com/gradle/gradle/blob/master/subprojects/base-services/src/main/java/org/gradle/util/Path.java , seems like what we need here

Reply:
It doesn't have validation. But without the user's input it's not needed.